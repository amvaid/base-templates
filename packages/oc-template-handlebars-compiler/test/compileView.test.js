jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000;

const path = require('path');
const fs = require('fs-extra');
const compileView = require('../lib/compileView.js');

const componentPath = path.join(
  __dirname,
  '../../../mocks/handlebars-component'
);
const publishPath = path.join(componentPath, '_package');
const publishFileName = 'template.js';

const options = {
  componentPackage: {
    oc: {
      files: {
        template: {
          src: 'template.hbs',
          type: 'handlebars'
        }
      }
    }
  },
  componentPath,
  publishPath,
  publishFileName
};

test('Should correctly compile the view', done => {
  compileView(options, (err, compiledViewInfo) => {
    expect(compiledViewInfo).toMatchSnapshot();
    expect(
      fs.readFileSync(path.join(publishPath, publishFileName), 'UTF8')
    ).toMatchSnapshot();
    fs.removeSync(path.join(publishPath, publishFileName));
    done();
  });
});
