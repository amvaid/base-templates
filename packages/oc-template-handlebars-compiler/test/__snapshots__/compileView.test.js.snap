// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the view 1`] = `
Object {
  "hashKey": "2112a7cce970d8e5eb8a60c5ada1c4d822b22b0e",
  "src": "template.js",
  "type": "handlebars",
}
`;

exports[`Should correctly compile the view 2`] = `"var oc=oc||{};oc.components=oc.components||{},oc.components[\\"2112a7cce970d8e5eb8a60c5ada1c4d822b22b0e\\"]={compiler:[7,\\">= 4.0.0\\"],main:function(c,o,e,n,a){return\\"<h1>Hello handlebars!</h1>\\"},useData:!0};"`;
